### ISSUE TITLE
API Implementation

### ISSUE THREAD

**Issue by MyIgel** [engelsystem/engelsystem#339]
Implement an API for user actions (Shifts, Profiledata etc.)

**Comment by Clevero** [comment]
Are there any news on this?

I've created an opensource volunteer app (iOS and Android) for a sports event that was taken over a weekend in Hessen Germany in June. Feedback was pretty good and I'm looking forward to developing the app in the future.
https://play.google.com/store/apps/details?id=me.ginga.lktf

This first version of the app had just the following functionalities:
* List of the FAQ
* List of the locations with further information like address, next bus stations (with links to the router finder of the local public transportation) and links to google maps
* List of your own shifts (where and when are they)
* contact persons
* reporting panel (feedback to your volunteer actions, event and app, are there physically broken things?)


The app was firstly developed directly using cordova and as the backend was a WordPress instance. My plan is it to reimplement it with Ionic, mostly for maintainability and design.

And I'm thinking the last couple of weeks if it would also be helpful for events where the volunteer management is done by the engelsystem. I'm interested in hopefully making a pull request to extend the existing API (the export of shifts for iCal or JSON) to also support features like getting locations, news, meetups and asking questions to the heaven

But I do not know if the MVC pattern is now fully implemented and if the plan for an API in the future was a different one.


Would be glad to hear some thoughts!

**Comment by MyIgel** [comment]
Hm, i created the issue as a ToDo for me while restructuring the engelsystem (see [here](https://github.com/engelsystem/engelsystem/projects/2), im currently doing exam stuff and after that I have to finish to migrate my server but after that I hope to make some progress with the engelsystem refactoring and after that adding new features like the API.

So the API depends on a redesigned backend what is currently a big "work in progress" :D
(That I hope to finish to the next congress)

-> Any thoughts about the design are welcome :)

**Comment by MyIgel** [comment]
I will add them :)

Contributions are welcome -> I think that we should talk about that soon :)
The current codebase is not the best starting point, thats the reason for my [rewrite](https://github.com/MyIgel/engelsystem/tree/rebuild) :)

So if you have some Ideas: Add a new issue / comment on existing ones -> I'm happy about all contributions

**Comment by Clevero** [comment]
Status update on my side:
I've rewritten the app using Ionic and added some new features to it and created a little site https://helfer-app.github.io for it (sorry, currently only in german). Hopefully I get some feedbacks at #35c3

<img src="https://helfer-app.github.io/Helfer-App%20Orte.gif" width="25%" height="25%" />
At the very top of my Roadmap is still integrating the engelsystem.
I'm not a professional programmer and also don't have experience creating API's that are relaible and well thought. So I don't think I could make a pull request for the API at the server side

Just as a status update. Doesn't want to bother you guys with that

**Comment by cbluoss** [comment]
Any updates regarding a (basic) API implementation?

**Comment by JasMich** [comment]
Yes, really needed. I would like to build a (Linux/*nix-) Desktop client.
